components/lsp/LSP.tsx

```diff
@@ -62,10 +61,11 @@ const Testing = () => {
       setShowSettle(true);
     }
   }, [contractExpirationTime, currentTime, contractState]);
+
   // Get contract data and set values.
   useEffect(() => {
-    if (signer && !lspContract) {
-      const contract = createLSPContractInstance(signer, KNOWN_LSP_ADDRESS);
+    if (signer && !lspContract && data.address && account) {
+      const contract = createLSPContractInstance(signer, data.address);
       contract.contractState().then(async (cs: ContractState) => {
```

Comment:
I beleive this exists on "data" as `contractState`

Reply:
Swapped over.